p2p/src/net/mock/backend.rs

```diff
@@ -252,13 +253,23 @@ where
     }
 
     async fn subscribe(&mut self, topics: BTreeSet<PubSubTopic>) {
-        let subscription = Box::new(Message::Subscribe { topics });
-        join_all(self.peers.iter().map(|(id, p)| {
-            p.tx.send(MockEvent::SendMessage(subscription.clone())).inspect_err(move |e| {
-                log::error!("Failed to send announcement to {id:?} peer: {e:?}")
+        let subscription = MockEvent::SendMessage(Box::new(Message::Subscribe { topics }));
+
+        // Send the message to all peers in pseudorandom order.
+        // TODO: This can be moved to a separate function and reused in the `announce_data`
+        // function, when we no longer need the special logic for the `InsufficientPeers` error.
+        let mut futures: Vec<_> = self
+            .peers
+            .iter()
+            .map(|(id, p)| {
+                p.tx.send(subscription.clone()).inspect_err(move |e| {
+                    log::error!("Failed to send subscription to {id:?} peer: {e:?}")
+                })
             })
-        }))
-        .await;
+            .collect();
+        futures.shuffle(&mut rand::thread_rng());
```

Comment:
Shouldn't `crypto::random::make_pseudo_rng()` be used instead?

Reply:
Sure! Fixed.